File: java/timesheet/src/main/java/timesheet/Freelancer.java
package timesheet;

public class Freelancer implements Worker {
    private String firstName;
    private String email;

    public Freelancer(String firstName, String email) {
        this.firstName = firstName;
        this.email = email;
    }

    @Override
    public String getFirstName() {
        return firstName;
    }

    @Override
    public String getEmail() {
        return email;
    }

    public void work() { /* Some freelance work */ }
    
    public Timesheet getTimesheet() {
        throw new UnsupportedOperationException("Freelancer doesn't support timesheets.");
    }
}

File: scala/timesheet/src/main/scala/timesheet/Freelancer.scala
package timesheet

class Freelancer(val firstName: String, val email: String) extends Worker {

  override def getFirstName: String = firstName

  override def getEmail: String = email

  override def work(): Unit = {
    // Some freelance work
  }

  override def getTimesheet: Timesheet = {
    throw new UnsupportedOperationException("Freelancer doesn't support timesheets.")
  }
}